taichi/analysis/value_diff.cpp

```diff
@@ -93,6 +93,49 @@ class ValueDiffLoopIndex : public IRVisitor {
   }
 };
 
+class ValueDiffPtrIndex : public IRVisitor {
```

Comment:
Although the motivation for this pass is to find the difference between `Stmt *` and a `LoopIndexStmt *`, the latter never appears in the code (except for the class name). This makes the code here a bit confusing. 

Maybe we should rename the class to `FindDirectValueBaseAndOffset` and add some comments?